backend/src/main/kotlin/org/loculus/backend/service/SequenceCompressionBackfillService.kt

```diff
@@ -0,0 +1,257 @@
+package org.loculus.backend.service.maintenance
+
+import mu.KotlinLogging
+import org.jetbrains.exposed.sql.JoinType
+import org.jetbrains.exposed.sql.SortOrder
+import org.jetbrains.exposed.sql.and
+import org.jetbrains.exposed.sql.andWhere
+import org.jetbrains.exposed.sql.or
+import org.jetbrains.exposed.sql.selectAll
+import org.jetbrains.exposed.sql.transactions.transaction
+import org.jetbrains.exposed.sql.update
+import org.loculus.backend.api.Organism
+import org.loculus.backend.api.OriginalData
+import org.loculus.backend.api.ProcessedData
+import org.loculus.backend.service.submission.CompressedSequence
+import org.loculus.backend.service.submission.CompressionDictService
+import org.loculus.backend.service.submission.SequenceEntriesPreprocessedDataTable
+import org.loculus.backend.service.submission.SequenceEntriesTable
+import org.springframework.scheduling.annotation.Scheduled
+import org.springframework.stereotype.Component
+import org.springframework.stereotype.Service
+
+private val log = KotlinLogging.logger {}
+
+@Component
+class SequenceCompressionBackfillStarter(private val backfill: SequenceCompressionBackfillService) {
+    @Scheduled(fixedDelay = 1, timeUnit = java.util.concurrent.TimeUnit.DAYS)
+    fun startAfterReady() {
+        backfill.run()
+    }
+}
+
+/**
+ * Backfills compressionDictId into JSONB blobs in sequence_entries.original_data
+ * and sequence_entries_preprocessed_data.processed_data.
+ *
+ */
+@Service
+class SequenceCompressionBackfillService(private val compressionDictService: CompressionDictService) {
+
+    fun run(batchSize: Int = 2_000, logEvery: Int = 1_000) {
+        log.info { "Backfill: START" }
+        val totalOriginal = backfillOriginalData(batchSize, logEvery)
+        val totalProcessed = backfillProcessedData(batchSize, logEvery)
+        log.info { "Backfill: DONE (originalData=$totalOriginal, processedData=$totalProcessed)" }
+    }
+
+    // Sequences without a compression DictId are compressed on a segment level (even original data)
+    private fun addCompressionDict(originalData: Map<String, CompressedSequence?>, organism: String) =
```

Comment:
`originalData` is a bit misleading as a name here, since it's also used as a separate type `OriginalData`.

```suggestion
    private fun addCompressionDict(sequences: Map<String, CompressedSequence?>, organism: String) =
```